File: _Uncategorized/1703_DetermineIfStringHalvesAreAlike.cs
// 1704. Determine if String Halves are Alike
// Tags: 
//

using System.Collections.Generic;

public class Solution 
{
    private static List<char> _vowels = new() {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};

    public bool HalvesAreAlike(string s) 
    {
        // Edge case.
        if(string.IsNullOrEmpty(s))
            return false;

        char[] ca = s.ToCharArray();
        int halfLength = ca.Length/2;

        int aCount = 0;
        int bCount = 0;

        // Count all of the vowels for both halves.
        for(int i = 0; i < halfLength; i++)
        {
            if(_vowels.Contains(ca[i]))
                aCount++;
            
            if(_vowels.Contains(ca[i + halfLength]))
                bCount++;
        }
        
        return aCount == bCount ? true : false;
    }
}

File: _Uncategorized/1703_DetermineIfStringHalvesAreAlike.rs
// 1704. Determine if String Halves are Alike
// Tags: 
//

impl Solution {
    const VOWELS: &'static [char; 10] = &['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];
    
    pub fn is_vowel(c: char) -> bool {
        
        for v in Solution::VOWELS {
            if (*v) == c {
                return true;
            }
        }
        false
    }
    
    pub fn halves_are_alike(s: String) -> bool {
        
        let chars: Vec<char> = s.chars().collect();
        let half_length = chars.len()/2;
        
        let mut a_count = 0;
        let mut b_count = 0;
        
        for i in 0..half_length {
            if Solution::is_vowel(chars[i]) {
                a_count += 1;
            }
            
            if Solution::is_vowel(chars[i + half_length]) {
                b_count += 1;
            }
        }
        
        a_count == b_count
    }
}